nutils/solver.py

```diff
@@ -436,11 +436,13 @@ def optimize(self, *, droptol: Optional[float], arguments: Dict[str, numpy.ndarr
             raise ValueError('system is not symmetric')
         x, iscons, arguments = self.prepare_solution_vector(arguments, constrain)
         jac, res, val = self.assemble(arguments)
-        nosupp = ~jac.rowsupp(droptol)
-        dx = -jac.solve(res, constrain=iscons | nosupp, **_copy_with_defaults(linargs, symmetric=self.is_symmetric))
+        mycons = iscons | ~jac.rowsupp(droptol)
+        dx = -jac.solve(res, constrain=mycons, **_copy_with_defaults(linargs, symmetric=self.is_symmetric))
         log.info(f'optimal value: {val+.5*(res@dx):.1e}') # val(x + dx) = val(x) + res(x) dx + .5 dx jac dx
         x += dx
-        x[nosupp & ~iscons] = numpy.nan
+        for trial, i, j in zip(self.trials, self.__trial_offsets, self.__trial_offsets[1:]):
+            log.info(f'constrained {(~mycons[i:j]).sum()} degrees of freedom of u')
```

Comment:
Instead of `u` the name of the trial argument should be formatted.

Perhaps we also want to see the total amount of dofs `j - i`? 'Constrained 10 of 200 degrees of freedom of u.'

Suggestion: `j - i - mycons[i:j].sum()` is faster and uses less memory at the expense of one non-whitespace character.

Reply:
fixed